src/runtime/threads/detail/thread_pool_base.cpp

```diff
@@ -95,7 +95,7 @@ namespace hpx { namespace threads { namespace detail
         auto const& rp = resource::get_partitioner();
         for (std::size_t i = 0; i != pool_threads; ++i)
         {
-            bit_or(used_processing_units_, rp.get_pu_mask(threads_offset + i));
+            used_processing_units_ |= rp.get_pu_mask(threads_offset + i);
```

Comment:
This may not work for all configurations. I think the correct fix would be:

    used_processing_units_ = bit_or(used_processing_units_, rp.get_pu_mask(threads_offset + i));

Reply:
No. 
```
bit_or(a,b)
```
returns a boolean true or false depending on the two masks and so it only sets the lowest bit. We actually want the true bitwise or operation and the submitted patch is correct. if `CPU's <= 64` then a `uint64_t` is bitwise or'ed (ok), if `CPUs>64`, then a `bitset<blah>` is or'ed.
I believe my patch is correct.

Reply:
Ok, thanks for pointing this out. Still. there is a third option, namely if we use `boost::dynamic_bitset`. I'm not sure if `operator|=()` is provided in that case. IIRC, it provides only `operator|()`.

Reply:
Sorry, I was mistaken about the small CPU count version.
`boost::dynamic_bitset` has the `!=` operator and it has the signature
`dynamic_bitset& operator|=(const dynamic_bitset& b);`
which is exactly what we want in this case.

Reply:
Ok. Thanks for clarifying.

Reply:
so we can merge then yes?

Reply:
Sure, could we still add that test, please?